registry/remote/repository.go

```diff
@@ -992,16 +1046,33 @@ func verifyContentDigest(resp *http.Response, expected digest.Digest) error {
 	if digestStr == "" {
 		return nil
 	}
-	contentDigest, err := digest.Parse(digestStr)
-	if err != nil {
+	if contentDigest, err := digest.Parse(digestStr); err != nil {
 		return fmt.Errorf("%s %q: invalid response Docker-Content-Digest: %s", resp.Request.Method, resp.Request.URL, digestStr)
-	}
-	if contentDigest != expected {
+	} else if contentDigest != expected {
 		return fmt.Errorf("%s: mismatch digest: %s", expected, contentDigest)
 	}
 	return nil
 }
 
+// verifyContentDigests is a superset of verifyContentDigest; it takes in the actual (calculated) digest, and
+// verifies against that, the other two digests, both of which may or may not be set to the empty digest.  If empty,
+// there is no validation error, however is not empty, it ensures that the given digest is valid (matches the actual
+// digest)
+func verifyContentDigests(actualDigest digest.Digest, digests map[string]digest.Digest) error {
+	for src, d := range digests {
+		if len(d.String()) == 0 {
```

Comment:
As `digest.Digest` is a string, `if len(d) == 0` is more efficient.

Reply:
This was more for communication to the reader, that while `d` may be a string, it's actually a `digest.Digest`.  If the change in `go-digest` I have open is shipped, we can instead use the abstract method `if d.IsEmpty`.  Not sure how much people on this project care about this abstraction that I'm talking about, since the counter argument is: "well we all know it's just a string".

I'll make it as per your suggestion meanwhile.  My preference would be `d.IsEmpty` because it is more communicative, but that's not available.  What are your thoughts?

Reply:
Yeah it would be good if there is `d.IsEmpty`. But before there is one, I think we can just keep it simple.